src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -158,10 +158,18 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
 
   // Components for collecting and generating output
 
-  /** Some statistics (normally disabled) set with -Ystatistics */
-  object statistics extends {
-    val global: Global.this.type = Global.this
-  } with StatisticsInfo
+  import scala.reflect.internal.util.StatisticsInfo
+  trait AllStats extends ReflectStats
+                    with TypersStats
+                    with ImplicitsStats
+                    with MacrosStats
+                    with BackendStats
+
+  override object statistics extends {
+    override type Global = Global.this.type
+    override val symbolTable: SymbolTable = Global.this
```

Comment:
maybe if you refine the type of `symbolTable` to `Global.this.type`, you don't need the `Global#CompilationUnit` type (and the `Global` abstract type) below (https://github.com/scala/scala/pull/6051/commits/08cd6c18aa290458efde61010e680636b18b48a2#r136072193)

Reply:
I tried this, and the compiler wasn't able to override it.